flink-streaming-java/src/main/java/org/apache/flink/streaming/api/datastream/KeyedStream.java

```diff
@@ -163,7 +166,11 @@ public KeyedStream(DataStream<T> dataStream, KeySelector<T, KEY> keySelector, Ty
 	 * @param size The size of the window.
 	 */
 	public WindowedStream<T, KEY, TimeWindow> timeWindow(AbstractTime size) {
-		return window(TumblingTimeWindows.of(size));
+		if (environment.getStreamTimeCharacteristic() == TimeCharacteristic.PROCESSING_TIME) {
```

Comment:
Yes, but I think this is inevitable because in the Streaming API we don't (yet) have deferred creation of the actual runtime operator graph from the graph of API operations as we have in the batch API.

I think there are some other cases where the assumption about the `ExecutionConfig` also break.